packages/packages.go

```diff
@@ -193,9 +200,44 @@ func (i *FileSystemIndexer) Init(ctx context.Context) (err error) {
 	if err != nil {
 		return fmt.Errorf("reading packages from filesystem failed: %w", err)
 	}
+
+	go i.watchPackageFileSystem(ctx)
 	return nil
 }
 
+func (i *FileSystemIndexer) watchPackageFileSystem(ctx context.Context) error {
+	if i.watchInterval == 0 {
+		i.logger.Debug(fmt.Sprintf("No watcher configured for %s indexer, packageList will not be updated", i.label))
+		return nil
+	}
+	i.logger.Debug(fmt.Sprintf("Watching %s indexer for changes", i.label))
+	ticker := time.NewTicker(i.watchInterval)
+	defer ticker.Stop()
+	for {
```

Comment:
Have you considered using file system notifications instead of an active loop? https://github.com/fsnotify/fsnotify

With this we would only read the files if they have actually changed, and as a future change we could even update the list of packages instead of rebuilding it, as we would know what has been added, removed or modified.

Though any this could be added later too, we will need the option of the active loop in any case for situations where notifications are not available.